bug(candle): --features candle alone fails to compile (MessageMetadata unresolved)

## Bug

`cargo build -p zeph-llm --features candle` succeeds, but `cargo nextest run -p zeph-llm --features candle --lib` fails with:

```
error[E0433]: failed to resolve: use of undeclared type `MessageMetadata`
```

The `candle` feature depends on `MessageMetadata` which is gated behind another feature flag not included when building `candle` alone.

## Impact

- Cannot run `zeph-llm` unit tests with `--features candle` in isolation
- CI runs with `--features full` which includes all features, masking this bug
- Prevents targeted candle-only test runs for classifier development (#2185)

## Reproduction

```bash
cargo nextest run -p zeph-llm --features candle --lib
# -> error[E0433]: failed to resolve: use of undeclared type MessageMetadata
```

Works fine with:
```bash
cargo nextest run --workspace --features full --lib  # all 6554 pass
```

## Fix

Identify which module declares `MessageMetadata` and ensure its feature gate is a dependency of the `candle` feature, or remove the dependency from candle-scoped test code.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -30,6 +30,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
 - fix(llm): `TriageRouter` now delegates `embed()` to the first embedding-capable tier provider instead of always returning `EmbeddingNotSupported`; `supports_embeddings()` reflects tier provider capability — resolves tool schema filter being silently disabled when `routing = "triage"` (#2174)
 - fix(core): `/provider` switch and `/provider status` now display the configured `name` field from `[[llm.providers]]` instead of the provider type string (e.g. `"openai"`); `active_provider_name` stored in `RuntimeConfig` and updated on every switch (#2173)
+- fix(llm): add missing `use crate::provider::MessageMetadata` import inside `#[cfg(test)]` in `candle_provider/template.rs`; `--features candle` alone now compiles and runs unit tests (`cargo nextest run -p zeph-llm --features candle --lib`) (#2189)
 - fix(mcp): narrow `new_directive` injection pattern to require colon suffix, preventing false positive match on legitimate phrases like "new persona" in Todoist MCP tool descriptions; add regression test (#2170)
 - fix(memory): run `PRAGMA wal_checkpoint(PASSIVE)` after FTS5 entity inserts to fix cross-session SYNAPSE seed lookup (#2166); checkpoint is called at `SqliteStore` startup (safety net) and after every `EntityResolver::resolve_batch` (targeted hook)
 - fix(config): add `[security.guardrail]` stub to `default.toml` so `--migrate-config` injects commented guardrail defaults for configs that have `[security]` but no `[security.guardrail]` (#2158)

--- a/crates/zeph-llm/src/candle_provider/template.rs
+++ b/crates/zeph-llm/src/candle_provider/template.rs
@@ -128,6 +128,7 @@ fn format_raw(messages: &[Message]) -> String {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::provider::MessageMetadata;
 
     fn sample_messages() -> Vec<Message> {
         vec![

